pstore/api.go

```diff
@@ -34,11 +37,108 @@ type Record struct {
 	Timestamp time.Time
 }
 
+// Writer is the interface for writing to persistent store.
+// Implementations of Writer must be safe to use with multiple goroutines.
 type Writer interface {
+	// Write writes given collection of records to persistent storage
+	Write(records []Record) error
+}
```

Comment:
I would be open to using pstore.RecordWriter and pstore.LimitedRecordWriter however as these names don't have the same redundancy.